tools/swift-demangle/swift-demangle.cpp

```diff
@@ -86,22 +102,45 @@ static void demangle(llvm::raw_ostream &os, llvm::StringRef name,
   }
 }
 
-static llvm::StringRef substrBefore(llvm::StringRef whole,
-                                    llvm::StringRef part) {
-  return whole.slice(0, part.data() - whole.data());
-}
+static int demangleSTDIN(const swift::Demangle::DemangleOptions &options) {
+  const ssize_t ChunkSize = 4096 * 4;
+  // This doesn't handle Unicode symbols, but maybe that's okay.
+  llvm::Regex maybeSymbol("_T[_a-zA-Z0-9$]+");
 
-static llvm::StringRef substrAfter(llvm::StringRef whole,
-                                   llvm::StringRef part) {
-  return whole.substr((part.data() - whole.data()) + part.size());
+  while (true) {
+    llvm::SmallString<ChunkSize> Buffer;
+    ssize_t ReadBytes;
+
+    Buffer.reserve(Buffer.size() + ChunkSize);
+    ReadBytes = read(STDIN_FILENO, Buffer.end(), ChunkSize);
```

Comment:
I'm concerned about breaking a symbol across chunk boundaries. Can we do something line-oriented instead? (I don't think we care about running this tool across binary data.)